scipy/spatial/distance.py

```diff
@@ -1468,6 +1419,7 @@ def pdist(X, metric='euclidean', p=2, w=None, V=None, VI=None):
         if w is not None and not mstr.startswith("test_"):
             if mstr in ['seuclidean', 'se', 's', 'mahalanobis']:
                 raise ValueError("metric %s incompatible with weights" % mstr)
+            # need to use python version for weighting
```

Comment:
Oh, the exception message can't be fixed. If you specify weights, you can't use seuclidean or mahalanobis. For them, you specify variance or inverse variance instead. If "test_mahalanobis" has distance specified, that will just fail later.